File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C/s378126458_comments_removed.c
#include <stdio.h>

int main() {
    int N, Q;
    scanf("%d %d", &N, &Q);
    char s[N+1];
    scanf("%s", s);
    int *pre = (int*)malloc((N + 1)*sizeof(int));
    for (int i = 0; i <= N; i++) pre[i] = 0;
    for (int i = 1; i < N; i++) {
        if (s[i - 1] == 'A' && s[i] == 'C') {
            pre[i] = 1;
        }
    }
    for (int i = 1; i < N; i++) {
        pre[i] += pre[i - 1];
    }
    while (Q--) {
        int l, r;
        scanf("%d %d", &l, &r);
        r--;
        printf("%d\n", pre[r] - pre[l - 1]);
    }
    free(pre);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Go/s378126458_comments_removed.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var N, Q int
	fmt.Scanf("%d %d", &N, &Q)
	s := make([]byte, N)
	fmt.Scanf("%s", &s)
	pre := make([]int, N+1)

	for i := 1; i < N; i++ {
		if s[i-1] == 'A' && s[i] == 'C' {
			pre[i] = 1
		}
	}

	for i := 1; i < N; i++ {
		pre[i] += pre[i-1]
	}

	for Q > 0 {
		var l, r int
		fmt.Scanf("%d %d", &l, &r)
		r--
		fmt.Println(pre[r] - pre[l-1])
		Q--
	}
	// End of Code
}
